pentagon/component/kops/__init__.py

```diff
@@ -0,0 +1,84 @@
+from pentagon.component import ComponentBase
+import os
+import glob
+import shutil
+import logging
+import traceback
+import sys
+import re
+import subprocess
+
+from pentagon.helpers import render_template
+
+
+class Cluster(ComponentBase):
+    _path = os.path.dirname(__file__)
+
+    def get(self, destination):
+
+        self._cluster_name = self._data.get('name', os.environ.get('CLUSTER_NAME'))
+        self._bucket = self._data.get('kops_state_store', os.environ.get('KOPS_STATE_STORE'))
+        self._destination = destination
+
+        if self._bucket is None:
+            logging.error("kops_state_store required.")
+            sys.exit()
+
+        if self._cluster_name is None:
+            logging.error("name is required.")
+            sys.exit()
+
+        os.mkdir(self._cluster_name)
+        os.chdir(self._cluster_name)
+
+        # get cluster yaml
+        logging.debug("Getting cluster.")
+        with open('cluster.yml', 'w') as cf:
+            args = ['kops',
+                    'get',
+                    'cluster',
+                    '--name={}'.format(self._cluster_name),
+                    '--state=s3://{}'.format(self._bucket),
+                    '-oyaml']
+            print " ".join(args)
+
+            subprocess.Popen(args, stdout=cf)
+
+        # get igs yaml
+        logging.debug("Getting instance groups.")
+        args = ['kops',
+                'get',
+                'ig',
+                '--name={}'.format(self._cluster_name),
+                '--state=s3://{}'.format(self._bucket)]
+
+        igs = [ig.split("\t") for ig in subprocess.check_output(args).split("\n")]
+        for ig in igs[1:-1]:
+            if "master" in ig[0]:
+                ig_file = open("master.yml", "a+")
+            else:
+                ig_file = open("{}.yml".format(ig[0]), "w")
```

Comment:
Please use a context manager for handling these files. If `subprocess.check_output` errors after this the file will remain open. This isn't a huge concern but is inconsistent with the other usage in this module and the more common Python-ism of using context managers for files.